OutSystems/outsystems-datagrid-reactive
# RGRIDT-721: Force the decimal places for decimal columns

This PR is for RGRIDT-721: Dirty marks are cleaned when we paste the original value

### What was happening
* Dirty marks are cleaned when we paste the original value
* Dirty marks are cleaned when we re-type the original value
* Decimal columns when autogenerated are being defined with 0 decimal places. For the 2 failing tests this is a problem because the cell on the grid has the value 99.90 for instance and the UI is showing 100. So after editing the value (to 120) and pasting the old one (100), the comparison 99.90 to 100 will inform us that the values are different and because of that the dirty mark will remain on the cell.

### What was done
* We added a new condition on the switch of the Column Generator. If the column is decimal, we want to force it to have 2 decimal places. Remember that this should only be a condition IF the columns are autogenerated!!

### Test Steps
1.Open the sample
**Expected**: Check if the price columns when not autogenerated have 2 decimal places.


1.Select a random cell of Price column
2.Press Ctrl+C
3.Type a new value over the cell, press enter
4.Select the same cell again
5.Press Ctrl+V
**Expected**: Cell isn't dirty- Row isn't dirty

1.Select a random cell of Price column
2.Remember its original value
3.Type a new value over the cell, press enter
4.Re-type its original value
**Expected**: Cell isn't dirty- Row isn't dirty


### Checklist
